pbj-integration-tests/src/test/java/com/hedera/pbj/intergration/test/Issue96Test.java

```diff
@@ -0,0 +1,281 @@
+package com.hedera.pbj.intergration.test;
+
+import com.google.protobuf.InvalidProtocolBufferException;
+import com.hedera.pbj.runtime.ProtoParserTools;
+import com.hedera.pbj.runtime.io.buffer.BufferedData;
+import com.hedera.pbj.runtime.io.buffer.Bytes;
+import com.hedera.pbj.test.proto.pbj.Everything;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Test;
+
+import java.util.HexFormat;
+
+import static org.junit.jupiter.api.Assertions.*;
+
+final class Issue96Test {
+
+    // ================================================================================================================
+    // Verify common comments.
+    @Test
+    @DisplayName("Issue 96")
+    void issue96() {
+        final var unhexed = HexFormat.of().parseHex(testData);
+        assertThrows(InvalidProtocolBufferException.class, () ->
+                com.hederahashgraph.api.proto.java.ServicesConfigurationList.parseFrom(unhexed));
+        assertThrows(Exception.class, () ->
+                com.hedera.hapi.node.base.ServicesConfigurationList.PROTOBUF.parseStrict(
+                        BufferedData.wrap(unhexed)));
+
+    }
+
+    @Test
+    @DisplayName("Test Boxed Float")
+    void testBoxedFloat() {
+        Everything ev = Everything.newBuilder().floatBoxed(123.0f).build();
+        Bytes bytes = Everything.PROTOBUF.toBytes(ev);
+        Bytes b = bytes.slice(0, bytes.length() - 1);
+        assertThrows(Exception.class, () ->
+                Everything.PROTOBUF.parseStrict(
+                        BufferedData.wrap(b.toByteArray())));
+    }
+
+    @Test
+    @DisplayName("Test Boxed Int32")
+    void testBoxedInt32() {
+        Everything ev = Everything.newBuilder().int32Boxed(123).build();
+        Bytes bytes = Everything.PROTOBUF.toBytes(ev);
+        Bytes b = bytes.slice(0, bytes.length() - 1);
+        assertThrows(Exception.class, () ->
+                Everything.PROTOBUF.parseStrict(
+                        BufferedData.wrap(b.toByteArray())));
+    }
+
+    @Test
+    @DisplayName("Test Boxed Int64")
+    void testBoxedInt64() {
+        Everything ev = Everything.newBuilder().int64Boxed(12345678L).build();
+        Bytes bytes = Everything.PROTOBUF.toBytes(ev);
+        Bytes b = bytes.slice(0, bytes.length() - 1);
+        assertThrows(Exception.class, () ->
+                Everything.PROTOBUF.parseStrict(
+                        BufferedData.wrap(b.toByteArray())));
+    }
+
+    @Test
+    @DisplayName("Test Number Int64")
+    void testNumberInt64() {
+        Everything ev = Everything.newBuilder().int32Number(12345678).build();
```

Comment:
It feels like it would be nice to add more tests of this style for all other field types possible. To make sure they are all good with various truncation lengths.

Reply:
Yes, I started with the Everything tests and had more, but then after debugging it turned out they all go through ProtoParserTools. All the types are handled by reads actually made in this class, so I thought that should cover everything. I can certainly add them back in. Just there was no real value added. Just let me know...